src/sklearn_evaluation/models.py

```diff
@@ -0,0 +1,430 @@
+        self.model = model
+        super().__init__()
+
+    def evaluate_balance(self, y_true):
+        """
+        Checks if model is balanced
+        """
+        balance_section = ReportSection("balance")
+
+        is_balanced = check_array_balance(y_true)
+
+        if is_balanced:
+            balance_section.set_is_ok(True)
+            balance_section.append_guideline("your model is balanced")
+        else:
+            balance_section.set_is_ok(False)
+            p = plot.target_analysis(y_true)
+            balance_section.append_guideline("Your test set is highly imbalanced")
+            balance_section.append_guideline(p)
+            balance_section.append_guideline(
+                "To tackle this, check out this "
+                "<a href='https://ploomber.io/blog/' target='_blank'>guide</a>"
+            )
+
+        self._add_section_to_report(balance_section)
+
+    def evaluate_accuracy(self, y_true, y_pred_test):
+        """
+        Measures how many labels the
+        model got right out of the total number of predictions
+        """
+        accuracy_threshold = 0.9
+        accuracy_section = ReportSection("accuracy")
+        accuracy = accuracy_score(y_true, y_pred_test)
+
+        balance = self.evaluation_state["balance"]
+
+        accuracy_section.append_guideline(f"Accuracy is {accuracy}")
+        if accuracy >= accuracy_threshold:
+            if balance["is_ok"]:
+                accuracy_section.append_guideline["is_ok"] = True
+                accuracy_section.append_guideline("You model is accurate")
+            else:
+                accuracy_section.set_is_ok(False)
+                accuracy_section.append_guideline(
+                    "Please note your model is unbalanced, "
+                    "so high accuracy could be misleading"
+                )
+
+        self._add_section_to_report(accuracy_section)
+
+    def evaluate_confusion_matrix(y_true, y_pred):
+        """
+        Checks how many of a classifier's predictions were correct,
+        and when incorrect, where the classifier got confused
+        """
+        # TODO: Implement
+        cm = confusion_matrix(y_true, y_pred)
+
+        # normalize
+        cm = cm.astype("float") / cm.sum(axis=1)[:, np.newaxis]
+
+        # TODO: Evaluate model by confusion matrix
+        diagonal = np.diagonal(cm)  # noqa
+
+    def evaluate_auc(self, y_true, y_score):
+        """
+        Checks if roc auc is in acceptable range
+        """
+        auc_section = ReportSection("auc")
+
+        auc_threshold_low_range = Range(0, 0.6)
+        auc_threshold_acceptable_range = Range(0.7, 0.8)
+
+        # auc - roc
+        roc = plot.ROC.from_raw_data(y_true, y_score)
+        for i in range(len(roc.fpr)):
+            roc_auc = auc(roc.fpr[i], roc.tpr[i])
+
+            # todo: better check
+            label = roc.label[i] if len(roc.label) > 0 else f"class {i}"
+            r = re.match(r"^\(class (.)*\)", label)
```

Comment:
let's open an issue on this one so we don't forget to tackle it, looks important